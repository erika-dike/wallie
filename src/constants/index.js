export const DEFAULT_PROFILE_PIC = 'http://res.cloudinary.com/andela-troupon/image/upload/v1491232845/default_profile_normal_n8yvkf.png';
