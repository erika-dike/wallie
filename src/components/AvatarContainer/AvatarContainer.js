import React from 'react';
import PropTypes from 'prop-types';
import {
  DropdownButton,
  MenuItem,
} from 'react-bootstrap';

import { DEFAULT_PROFILE_PIC } from '../../constants/';


const getTitle = img =>
  <a
    className="ProfileCard-avatar-link ProfileAvatar-placeholder u-inline-block profile-picture"
    href="edit-image"
    tabIndex="-1"
    aria-hidden="true"
    role="button"
    onClick={e => e.preventDefault()}
  >
    {
      img === DEFAULT_PROFILE_PIC
      ?
        <div
          className="ProfileCard-avatar-image u-border-user-color-light-hover u-bg-user-color"
        >
          <span className="Icon Icon--extra-large">
            <i className="fa fa-camera" aria-hidden="true" />
          </span>
        </div>
      :
        <img
          className="ProfileCard-avatar-image js-action-profile-avatar"
          src={img}
          alt="Avatar"
        />
    }
  </a>;

const AvatarContainer = ({ img }) =>
  <div className="ProfileCard-avatar-container">
    <div id="choose-photo" className="contols avatar-settings inline-upload-avatar dropdown center">
      <DropdownButton bsStyle="link" title={getTitle(img)} id="dropdown-test" noCaret>
        <MenuItem eventKey="1">Upload Photo</MenuItem>
        <MenuItem eventKey="2">Change Photo</MenuItem>
        <MenuItem divider />
        <MenuItem eventKey="4">Cancel</MenuItem>
      </DropdownButton>
    </div>
  </div>;

AvatarContainer.defaultProps = {
  img: DEFAULT_PROFILE_PIC,
};

AvatarContainer.propTypes = {
  img: PropTypes.string,
};

export default AvatarContainer;
