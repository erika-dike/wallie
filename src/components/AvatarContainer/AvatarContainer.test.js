import React from 'react';
import { mount } from 'enzyme';
import { MenuItem } from 'react-bootstrap';
import renderer from 'react-test-renderer';

import { DEFAULT_PROFILE_PIC } from '../../constants/';

import AvatarContainer from './AvatarContainer';


describe('AvatarContainer Component Test Suite', () => {
  let props;

  beforeEach(() => {
    props = {
      img: 'http://fake-pics.jpg',
      uploadImage: jest.fn(),
    };
  });

  it('renders correctly', () => {
    const tree = renderer.create(<AvatarContainer {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('uses default image url if none is provided for user', () => {
    props.img = undefined;
    const wrapper = mount(<AvatarContainer {...props} />);
    expect(wrapper.prop('img')).toEqual(DEFAULT_PROFILE_PIC);
  });

  it('renders camera if no image is provided', () => {
    const wrapper = mount(<AvatarContainer {...props} />);
    expect(wrapper.find('.fa-camera').exists()).toBeFalsy();
    wrapper.setProps({ img: undefined });
    expect(wrapper.find('.fa-camera').exists()).toBeTruthy();
  });

  it('calls uploadImage function when MenuItem with eventKey 1 is clicked', () => {
    const wrapper = mount(<AvatarContainer {...props} />);
    const uploadImageLink = wrapper.find(MenuItem).at(0).find('a');
    uploadImageLink.simulate('click');
    expect(props.uploadImage).toHaveBeenCalled();
  });
});
