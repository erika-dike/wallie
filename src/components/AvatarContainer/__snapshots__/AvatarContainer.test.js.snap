exports[`AvatarContainer Component Test Suite renders correctly 1`] = `
<div
  className="ProfileCard-avatar-container">
  <div
    className="contols avatar-settings inline-upload-avatar dropdown center"
    id="choose-photo">
    <div
      className="dropdown btn-group btn-group-link">
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-link"
        disabled={false}
        id="dropdown-test"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button">
        <a
          aria-hidden="true"
          className="ProfileCard-avatar-link ProfileAvatar-placeholder u-inline-block profile-picture"
          href="edit-image"
          onClick={[Function]}
          role="button"
          tabIndex="-1">
          <img
            alt="Avatar"
            className="ProfileCard-avatar-image js-action-profile-avatar"
            src="http://fake-pics.jpg" />
        </a>
      </button>
      <ul
        aria-labelledby="dropdown-test"
        className="dropdown-menu"
        role="menu">
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1">
            Upload Photo
          </a>
        </li>
        <li
          className="divider"
          onKeyDown={[Function]}
          role="separator"
          style={undefined} />
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1">
            Cancel
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
